data/src/main/java/io/gnosis/data/repositories/CredentialsRepository.kt

```diff
@@ -0,0 +1,74 @@
+package io.gnosis.data.repositories
+
+import io.gnosis.data.db.daos.OwnerDao
+import io.gnosis.data.models.Owner
+import io.gnosis.data.security.HeimdallEncryptionManager
+import io.gnosis.data.utils.toSignatureString
+import kotlinx.coroutines.runBlocking
+import pm.gnosis.crypto.KeyPair
+import pm.gnosis.model.Solidity
+import pm.gnosis.svalinn.security.EncryptionManager
+import pm.gnosis.svalinn.security.db.EncryptedByteArray
+import java.math.BigInteger
+
+class CredentialsRepository(
+    private val ownerDao: OwnerDao,
+    private val encryptionManager: HeimdallEncryptionManager,
+    //FIXME: remove after all users migrate to version with db storage for owners
+    private val ownerVault: OwnerCredentialsRepository
+) {
+
+    init {
+        runBlocking {
+            if (ownerVault.hasCredentials()) {
+                val credentials = ownerVault.retrieveCredentials()!!
+                saveOwner(credentials.address, credentials.key)
+                ownerVault.removeCredentials()
+            }
+        }
+    }
+
+    suspend fun ownerCount(): Int {
+        return ownerDao.ownerCount()
+    }
+
+    suspend fun owners(): List<Owner> {
+        return ownerDao.loadAll()
+    }
+
+    suspend fun saveOwner(
+        address: Solidity.Address,
+        key: BigInteger,
+        name: String? = null
+    ) {
+        encryptionManager.unlock()
+        val owner = Owner(
+            address = address,
+            name = name,
+            type = Owner.Type.LOCALLY_STORED,
+            privateKey = EncryptedByteArray.create(encryptionManager, key.toByteArray())
+        )
+        encryptionManager.lock()
+        ownerDao.save(owner)
+    }
+
+    suspend fun saveOwner(owner: Owner) {
+        ownerDao.save(owner)
+    }
+
+    suspend fun removeOwner(owner: Owner) {
+        ownerDao.delete(owner)
+    }
+
+    fun signWithOwner(owner: Owner, data: ByteArray): String {
+        val cryptoData = EncryptionManager.CryptoData.fromString(owner.privateKey.toString())
```

Comment:
We need to use:
``` 
val converter = EncryptedByteArray.Converter()
val cryptoData = EncryptionManager.CryptoData.fromString(converter.toStorage(owner.privateKey))
```

Instead of `toString()`

It looks like the converter can be reused.

Reply:
Maybe we can even get rid of this to String -> from String conversion?